modules/otel-collector/main.tf

```diff
@@ -0,0 +1,225 @@
+          processors = [
+            "memory_limiter",
+            "batch"
+          ],
+          receivers = [
+            "otlp"
+          ]
+        }
+      }
+    }
+  }
+
+  gcp_relay_conf = {
+    exporters = {
+      googlecloud = {}
+    }
+    extensions = {
+      health_check = {}
+    }
+    processors = {
+      batch = {}
+      memory_limiter = {
+        ballast_size_mib = 100
+        check_interval   = "5s"
+        limit_mib        = 200
+        spike_limit_mib  = 62
+      }
+    }
+    receivers = {
+      otlp = {
+        protocols = {
+          grpc = null
+          http = null
+        }
+      }
+    }
+    service = {
+      extensions = [
+        "health_check"
+      ]
+      pipelines = {
+        traces = {
+          receivers = [
+            "otlp"
+          ]
+          processors = [
+            "memory_limiter",
+            "batch"
+          ]
+          exporters = [
+            "googlecloud"
+          ]
+        }
+        metrics = {
+          receivers = [
+            "otlp"
+          ]
+          processors = [
+            "memory_limiter",
+            "batch"
+          ]
+          exporters = [
+            "googlecloud"
+          ]
+        }
+      }
+    }
+  }
+
+  relay_config = var.cloud_provider == "aws" ? local.aws_relay_conf : (var.cloud_provider == "gcp" ? local.gcp_relay_conf : null)
+
+  atomic           = var.atomic != null ? var.atomic : true
+  chart_name       = var.chart_name != null ? var.chart_name : "opentelemetry-collector"
+  chart_repository = var.chart_repository != null ? var.chart_repository : "https://open-telemetry.github.io/opentelemetry-helm-charts"
+  chart_version    = var.chart_version != null ? var.chart_version : "0.8.0"
+  cleanup_on_fail  = var.cleanup_on_fail != null ? var.cleanup_on_fail : true
+  create_namespace = var.create_namespace != null ? var.create_namespace : true
+  image_version    = var.image_version != null ? var.image_version : "0.40.0"
+  namespace        = var.namespace != null ? var.namespace : "sn-system"
+  release_name     = var.release_name != null ? var.release_name : "otel-collector"
+  settings         = var.settings != null ? var.settings : {}
+  timeout          = var.timeout != null ? var.timeout : 120
+
+  values = var.values != null ? var.values : yamlencode({
+    fullNameOverride = "opentelemetry-collector"
+    nameOverride     = "opentelemetry-collector"
+    agentCollector = {
+      enabled = false
+    }
+    autoscaling = {
+      enabled                           = true
+      minReplicas                       = 1
+      maxReplicas                       = 10
+      targetCPUUtilizationPercentage    = 80
+      targetMemoryUtilizationPercentage = 80
+    }
+    image = {
+      tag = local.image_version
+    }
+    standaloneCollector = {
+      enabled = true
+      resources = {
+        limits = {
+          cpu    = "500m"
+          memory = "256M"
+        }
+      }
+      # configOverride = {
```

Comment:
I don't understand why you're trying to configure anything using `configOverride`.  I believe that the chart creates the configmap, and we can drop the `otel_relay_configmap`.

Seems to me that the receivers (zipkin, etc) will be the same in all deployments, and the exporters will vary by cloud.

Reply:
The advantage of using the `configOverride` value in the helm chart, in any scenario (`sn-bootstrap` module vs composed in `terraform-aws-cloud`), would be the ability for us to do a one-shot apply. If we manage the configmap separately (i.e.`resource "kubernetes_manifest" "otel_relay_configmap" {}`), we would still need to do it _after_ the cluster is created.

Generally, your suggestion around making this cloud provider agnostic makes sense. I'll plan on removing the `cloud_provider` logic before merging this PR, so this module can just be composed in to the respective `terraform-*-cloud` module(s) for any special exporter config. 

In particular, if I can get Helm to feel good when using the the `configOverride` value, making that an input parameter into the module is probably the best way forward.